src/draw.jl

```diff
@@ -149,16 +149,41 @@ end
 function spannable_xy_labels(layout)
     labs = map(layout.content) do _ax
         ax = _ax.content
-        (x = ax.xlabel[], y = ax.ylabel[], empty = isemptyax(ax))
+        (x = ax.xlabel[], y = ax.ylabel[], 
+         xpad = ax.xlabelpadding[], ypad = ax.ylabelpadding[], empty = isemptyax(ax))
     end |> StructArray
     
-    unique_x_labs = unique(labs.x[.! labs.empty])
-    unique_y_labs = unique(labs.y[.! labs.empty])
-    
-    xlab = length(unique_x_labs) == 1 ? only(unique_x_labs) : nothing
-    ylab = length(unique_y_labs) == 1 ? only(unique_y_labs) : nothing
+    # if layout has multiple columns, check if xlabel is spannable
+    if size(layout)[2] > 1
+        unique_x_labs = unique(labs.x[.! labs.empty])
+        xlab = length(unique_x_labs) == 1 ? only(unique_x_labs) : nothing
+        if !isnothing(xlab)
+            unique_x_pads = unique(labs.xpad[.! labs.empty])
+            xpad = only(unique_x_pads)
+        else
+            xpad = nothing
+        end  
+    else
+        xlab = nothing
+        xpad = nothing
+    end
     
-    (x = xlab, y = ylab)
+    # if layout has multiple rows, check if xlabel is spannable
+    if size(layout)[1] > 1
+        unique_y_labs = unique(labs.y[.! labs.empty])
+        ylab = length(unique_y_labs) == 1 ? only(unique_y_labs) : nothing
+        if !isnothing(ylab)
+            unique_y_pads = unique(labs.ypad[.! labs.empty])
+            ypad = only(unique_y_pads)
+        else
+            ypad = nothing
+        end  
+    else
+        ylab = nothing
+        ypad = nothing
+    end
+        
+    (x = (lab=xlab, pad=xpad), y = (lab=ylab, pad=ypad))
```

Comment:
I think it's clearer without abbreviations, that is `(label=xlab, padding=xpad)`.